src/classes/CON_ContactMerge_CTRL.cls

```diff
@@ -274,6 +274,14 @@ public with sharing class CON_ContactMerge_CTRL {
             Schema.SobjectField f = contactFields.get(fieldName);
             Schema.DescribeFieldResult fResult = f.getDescribe();
             
+            // Ignore fields that have an identical API name as an npsp prefixed field
+            // This is due to a bug in dynamic SOQL, quich causes querying for two real fields, 
+            // e.g. npsp__foo__c and foo__c, to throw a duplicate field selected error
+            if (fieldName != UTIL_Namespace.StrTokenRemoveNSPrefix(fieldName) && 
+                contactFields.containsKey(UTIL_Namespace.StrTokenRemoveNSPrefix(fieldName))) {
+                continue;
+            }
+
```

Comment:
Just so I'm clear, you are basically not adding to the soql the npsp__foo__c field if  they have an unmanaged foo__c field.  But which one will then get displayed & merged.  i think it will be their field, rather than the npsp one.  and this will happen regardless of what they have in any field set.  if they have a shadow field, it will be the one copied.  did I understand this correctly?  If so, I kinda hate that even if they put our npsp field in the field set, we will still end up only copying values between their shadow fields.  not sure on how to solve this though.  I'd prefer it to be if you have a field that tries to shadow the npsp field, we will ignore your field and use the npsp field.